xpass/xpass.h

```diff
@@ -176,8 +207,10 @@ class XPassAgent: public Agent {
   void recv_credit(Packet *pkt);
   void recv_data(Packet *pkt);
   void recv_credit_stop(Packet *pkt);
+  void recv_nack(Packet *pkt);
 
   void handle_retransmit();
+  void handle_retransmit_credit_stop();
```

Comment:
This also can be combined into handle_retransmit according to the credit_recv_state_.